src/job-manager/plugins/delegate.c

```diff
@@ -29,20 +87,55 @@ static int depend_cb (flux_plugin_t *p,
                       flux_plugin_arg_t *args,
                       void *arg)
 {
-    flux_jobid_t id;
-    const char *s;
-
+    flux_t *h = flux_jobtap_get_flux (p);
+    flux_jobid_t *id;
+    flux_t *delegated;
+    const char *uri;
+    json_t *jobspec;
+    char *encoded_jobspec = NULL;
+    flux_future_t *jobid_future = NULL;
 
+    if (!h || !(id = malloc (sizeof (flux_jobid_t)))) {
+        return flux_jobtap_reject_job (p,
+                                       args,
+                                       "error processing delegate: %s",
+                                       flux_plugin_arg_strerror (args));
+    }
     if (flux_plugin_arg_unpack (args,
                                 FLUX_PLUGIN_ARG_IN,
-                                "{s:I s:{s:s}}",
-                                "id", &id,
+                                "{s:I s:{s:s} s:o}",
+                                "id", id,
                                 "dependency",
-                                  "value", &s) < 0)
+                                  "value", &uri, "jobspec", &jobspec) < 0
+        || flux_jobtap_job_aux_set (p, *id, "flux::jobid", id, free) < 0) {
+        free (id);
         return flux_jobtap_reject_job (p,
                                        args,
                                        "error processing delegate: %s",
                                        flux_plugin_arg_strerror (args));
+    }
+    if (!(delegated = flux_open (uri, 0))){
+        flux_log_error (h, "%" PRIu64 ": could not open URI %s", *id, uri);
+        return -1;
+    }
```

Comment:
Per @jameshcorbett's comment, I think you'll want to call [flux_set_reactor(3)](https://flux-framework.readthedocs.io/projects/flux-core/en/stable/man3/flux_get_reactor.html) here so that the job manager reactor services the new flux_t handle as well, e.g.

```c
if (flux_set_reactor (delegated, flux_get_reactor (h))) < 0)
    //handle error
```

Reply:
@garlick it seems the instance that `delegated` is a handle for is not receiving the job being sent, and the submission callback never fires. Do you know what's going wrong? I added the `flux_set_reactor ()` call you mentioned here and I also changed the timeout on the `flux_future_then ()` call to be negative, down from `0.0` (I thought maybe that was affecting it, but it seems not).

It's puzzling to me and I don't really know how to debug, and I was hoping you might see the issue right away.

Reply:
@jameshcorbett Want to debug together some time today or tomorrow?

Reply:
@garlick I'm not sure what the expected behavior of that command is but in the delegated instance it appears to hang. 

I have the jobtap plugin loaded in a 3-node allocation on fluke, and I have a 1-node subinstance (jobid fitH8BQj) launched via `flux batch` within my 3-node instance.

```
[corbett8@fluke128:flux-multi-cluster-utilities]$ flux proxy fSFVwYsZ flux submit -n1 hostname
fitH8BQj
[corbett8@fluke128:flux-multi-cluster-utilities]$ flux proxy fSFVwYsZ flux jobs
       JOBID USER     NAME       ST NTASKS NNODES     TIME INFO
[corbett8@fluke128:flux-multi-cluster-utilities]$ flux proxy fSFVwYsZ flux jobs -aA
       JOBID USER     NAME       ST NTASKS NNODES     TIME INFO
    fitH8BQj corbett8 hostname   CD      1      1   0.107s fluke130
[corbett8@fluke128:flux-multi-cluster-utilities]$ flux proxy fSFVwYsZ flux module trace job-ingest 
<hang>
```

Reply:
I think you may need to have the trace running at the time the job is submitted to the subinstance.

I'd also add some logging (after the `flux_open()` to the delegate, and after submitting the job) just as positive feedback that at least the jobtap plugin is progressing. (oh I guess you can know if the jobtap plugin progressed if the dependency has been added to the eventlog?)

Reply:
Yeah the dependency is added to the eventlog and I already tried adding some log messages at the end of the function, and they do print. I'll try having the trace running when submitting the job.

Reply:
< I think you may need to have the trace running at the time the job is submitted to the subinstance

Yep.  And the command will block waiting for messages.  It runs until you stop it.

Reply:
Next I'd try opening the delegate handle with `FLUX_O_TRACE` which should dump a trace of all messages on that handle to stderr. See if you get anything with that.

Reply:
I see now what the problem is, I should have seen this coming:
```
< 0.30239
< job-ingest.submit
< flags=topic,payload,route userid=31193 rolemask=owner,local errnum=22 matchtag=1
< dependency scheme "delegate" not supported
```

However I'm still puzzled why the job submission future's callback didn't fire? I need some way to be informed that the job was rejected.

Reply:
Hmm.  Just to double check, is `flux_set_reactor(delegated)` being called right after `delegated = flux_open()`?

If you log a message in `submit_callback()` before the first conditional, nothing?

Reply:
I figured out the issue, just another silly mistake on my part :) it seems to be working now. Thanks for the help!

Reply:
Whew :-)  Glad everything is working as designed.